backend/src/routers/api.ts

```diff
@@ -21,10 +21,20 @@ apiRouter.post(
 
             try {
                 //Validate request and collect fields from ApiRequest
-                const { id, prompt, contexts } = await validateApiRequest(body);
+                const { id, prompt, contexts, parameters } =
+                    await validateApiRequest(body);
 
                 //Retrieve GPT3 response from proxy, and create our response based on it
-                const gpt = await sendToProxy(createPrompt(contexts, prompt));
+                const gpt = await sendToProxy(
+                    createPrompt(
+                        contexts,
+                        prompt,
+                        parameters.creativity,
+                        parameters.quality,
+                        parameters.inputLength,
+                        parameters.outputLength
```

Comment:
(This is "improvement" is opinionated)
When function has more than couple parameters, the object should be given as parameter instead of multiple parameters, e.g.,
```javascript
function foo({ first: val1, second: val2, third: val3}) {}
````
instead of
```javascript
function bar(val1, val2, val3) {}
```
because in the object, the order of the functions is not fixed and user can not give accidentally parameters in wrong order.
e.g.,
```javascript
function bar(val1, val2, val3) {}
````

This way we don't introduce possibility of a bug caused by wrong parameter order which can be hard to spot in the code.